ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/CopyLocationAction.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ *  Copyright (c) 2023 Eclipse Contributors and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *
+ *******************************************************************************/
+package org.eclipse.pde.internal.ui.shared.target;
+
+import java.nio.file.Path;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.viewers.StructuredViewer;
+import org.eclipse.pde.core.target.TargetBundle;
+import org.eclipse.pde.core.target.TargetFeature;
+import org.eclipse.swt.dnd.Clipboard;
+import org.eclipse.swt.dnd.TextTransfer;
+import org.eclipse.swt.dnd.Transfer;
+import org.eclipse.ui.ISharedImages;
+import org.eclipse.ui.PlatformUI;
+
+class CopyLocationAction extends Action {
+
+	private StructuredViewer fViewer;
+
+	private String fLocation;
+
+	public CopyLocationAction(StructuredViewer viewer) {
+		fViewer = viewer;
+
+		setText(Messages.CopyLocationAction_copyLocation);
+		ISharedImages workbenchImages = PlatformUI.getWorkbench().getSharedImages();
+		setImageDescriptor(workbenchImages.getImageDescriptor(ISharedImages.IMG_TOOL_COPY));
+		setDisabledImageDescriptor(workbenchImages.getImageDescriptor(ISharedImages.IMG_TOOL_COPY_DISABLED));
+
+		viewer.addPostSelectionChangedListener(event -> {
+			fLocation = null;
+			var selection = event.getStructuredSelection();
```

Comment:
Is there a specific reason why you used `var` here and below instead of the actual type?
Personally I would have just used the type, since it isn't a complex generics chain here and I would say that it is easier to understand with just the type written out.

Reply:
Oh, oh.  We need a var-policy!  To use or when to use, that is the question?  😶

Reply:
I will change it if you wish...

Reply:
> Oh, oh. We need a var-policy! To use or when to use, that is the question? 😶

That would be good, yes. 😄
Personally I only use var when the type is too 'long' (e.g. a complex, nested generic type) or something internal that generates a discouraged-access warning. Then one can often use the fully-qualified-name in the constructor and `var` for the type and suppress the warning for the variable and often get that in one line:
`var name = new org.eclipse.foo.bar.internal.ClassName()`
The downside of var is that one cannot read the type from the variable declaration anymore. If one calls a constructor on the RHS of the assignment it is still simple to figure out, but the return type of methods is often hard/impossible to guess.
So I only use war in the cases above or if the type is really, really uninteresting.

But that's just my personal preference, but I think it would be good to have a general agreement about the usage of var in the Eclipse SDK. Since we don't have that now, I would leave that to your discretion since it is your change :)